common-util/functions/frontend-library.js

```diff
@@ -4,17 +4,30 @@ import {
   getChainIdOrDefaultToMainnet as getChainIdOrDefaultToMainnetFn,
   sendTransaction as sendTransactionFn,
   LOCAL_FORK_ID,
+  notifyError,
 } from '@autonolas/frontend-library';
 
 import { RPC_URLS } from 'common-util/constants/rpcs';
 import { SUPPORTED_CHAINS } from 'common-util/config/wagmi';
+import { isString } from 'lodash';
 
 const supportedChains =
   process.env.NEXT_PUBLIC_IS_CONNECTED_TO_LOCAL === 'true'
     ? [...SUPPORTED_CHAINS, { id: LOCAL_FORK_ID }]
     : SUPPORTED_CHAINS;
 
-export const getProvider = () => getProviderFn(supportedChains, RPC_URLS);
+export const getProvider = () => {
+  const provider = getProviderFn(supportedChains, RPC_URLS);
+  // not connected, return fallback RPC URL
+  if (isString(provider)) return provider;
```

Comment:
can we just do `typeof provider  === 'string'` instead of importing lodash?

Reply:
had a look, lodash does some more concrete checks: https://github.com/lodash/lodash/blob/main/src/isString.ts

both are options are fine, but will change to `typeof` as they're not really necessary here